lib/util/PDF/shareAndDownload.dart

```diff
@@ -317,10 +23,49 @@ Future<void> downloadPDF(List<dynamic> titles, List<dynamic> subTitles,
   List<String> phoneNumbers =
       prefs.getStringList('PhonePageSavedPhoneNumbers') ?? [];
   String username = prefs.getString('name') ?? '';
+  return {
+    'DifficultEvents': userSelectionDifficultEvents,
+    'MakeSafer': userSelectionMakeSafer,
+    'FeelBetter': userSelectionFeelBetter,
+    'Distractions': userSelectionDistractions,
+    'phoneNames': phoneNames,
+    'phoneNumbers': phoneNumbers,
+    'username': username
+  };
+}
+
+List<List<String>> filterEmptyData(List<List<String>> data) {
+  List<List<String>> realData = [];
+  for (var i = 0; i < data.length; i++) {
+    if (data[i].isEmpty) {
+      continue;
+    }
+    realData.add(data[i]);
+  }
+  return realData;
+}
+
+Future<pw.Document> createPDF(List<dynamic> titles, List<dynamic> subTitles,
+    Map<String, String> texts) async {
+  final pageFormat = PdfPageFormat.a4; // Set the page format to A4
+
+  // Load the font for the PDF
+  final ByteData fontData = await rootBundle.load('assets/fonts/CALIBRI.TTF');
```

Comment:
It's best not to use flutter related packages in non-ui code as it will improve usability and make our life easier on future changes. Can't you use dart only packages?

Reply:
I started doing that but didn't find the immediate dart packages for some of the things used from flutter, and didn't want to invest too much time on that. Specifically, did not find where/how to get the dart implementation of  "rootBundle.load".(Not that I invested more than 20 min on that in total, but still)